sklearn/_isotonic.pyx

```diff
@@ -75,10 +75,10 @@ def _make_unique(cnp.ndarray[dtype=floating] X,
     """
     unique_values = len(np.unique(X))
 
-    cdef cnp.ndarray[dtype=floating] y_out = np.empty(unique_values,
+    cdef floating[::1] y_out = np.empty(unique_values,
```

Comment:
`X.dtype` is not avalible if `X` is a memoryview. One needs to get the dtype directly:

```python
    if floating is float:
        dtype = np.float32
    else:
        dtype = np.float64

    cdef floating[::1] y_out = np.empty(unique_values, dtype=dtype)
```

We also need to change the following:

https://github.com/scikit-learn/scikit-learn/blob/3d26e28506d632c1c376a995eb553aa9be6275d0/sklearn/_isotonic.pyx#L91

Reply:
``cdef floating eps = np.finfo(X.dtype).resolution`` 
can you suggest what I need to change?

Reply:
Use the newly defined `dtype`

```python
cdef floating eps = np.finfo(dtype).resolution
```